app/lib/actions/katello/repository/sync.rb

```diff
@@ -16,27 +16,37 @@ class Sync < Actions::EntryAction
         # @param pulp_sync_task_id in case the sync was triggered outside
         #   of Katello and we just need to finish the rest of the orchestration
         # @param source_url optional url to override source URL with
-        def plan(repo, pulp_sync_task_id = nil, source_url = nil, incremental = false)
+        def plan(repo, pulp_sync_task_id = nil, options = {})
           action_subject(repo)
 
-          if repo.url.blank? && source_url.blank?
-            fail _("Unable to sync repo. This repository does not have a feed url.")
-          end
+          source_url = options.fetch(:source_url, nil)
+          incremental = options.fetch(:incremental, false)
+          validate_contents = options.fetch(:validate_contents, false)
+          skip_metadata_check = options.fetch(:skip_metadata_check, false)
+
+          pulp_sync_options = {}
+          pulp_sync_options[:download_policy] = ::Runcible::Models::YumImporter::DOWNLOAD_ON_DEMAND if validate_contents
+          pulp_sync_options[:force_full] = true if validate_contents || skip_metadata_check
+          pulp_sync_options[:remove_missing] = false if incremental
+
+          fail _("Unable to sync repo. This repository does not have a feed url.") if repo.url.blank? && source_url.blank?
+          fail _("Cannot validate contents on non-yum repositories.") if validate_contents && !repo.yum?
+          fail _("Cannot skip metadata check on non-yum repositories.") if skip_metadata_check && !repo.yum?
```

Comment:
Hrmm, will the controller propagate these to the user with a proper HTTP code? I can test, but I have this feeling these would result in 500s which would be overkill.